include/lbann/layers/io/input/generic_input_layer.hpp

```diff
@@ -29,11 +29,19 @@
 
 #include "lbann/layers/io/io_layer.hpp"
 //#include "lbann/utils/dataset.hpp"
+#include "lbann/io/persist.hpp"
 #include "lbann/io/data_buffers/generic_io_buffer.hpp"
 #include "lbann/io/data_buffers/partitioned_io_buffer.hpp"
 #include "lbann/models/model.hpp"
 #include "lbann/callbacks/imcomm.hpp"
 #include "lbann/utils/omp_diagnostics.hpp"
+#include "lbann/training_algorithms/training_algorithm.hpp"
+#include "lbann/training_algorithms/sgd_training_algorithm.hpp"
```

Comment:
I think this is another case of just needing `execution_context`, not the full training algorithms (if I'm wrong, I have more serious questions about why an input layer needs to know about the training algorithm since its job is just to provide input...)